influxql/scanner.go

```diff
@@ -596,6 +600,7 @@ func SplitIdent(s string) (segments []string, err error) {
 		if ch, _, err := r.ReadRune(); err != nil {
 			return segments, nil
 		} else if ch == '.' {
+			r.UnreadRune()
```

Comment:
Actually, I think the `SplitIdent` function can be deleted after I make one more small change.